From 3d0c15678f85eb7b24d9d1f949bef3c45d5f1352 Mon Sep 17 00:00:00 2001
From: Matthew Schwartz <matthew.schwartz@linux.dev>
Date: Sat, 30 Nov 2024 17:27:42 -0800
Subject: [PATCH] [util]/[hack]: disable VK_KHR_present_wait under all Nvidia
 drivers

Fixes presentation in some games that use DX11 freezing under gamescope,
only when VK_KHR_present_wait is in use on NV. This is a temporary
workaround until a proper fix is found.

Link: https://github.com/ValveSoftware/gamescope/issues/1592
---
 src/dxvk/dxvk_adapter.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/dxvk/dxvk_adapter.cpp b/src/dxvk/dxvk_adapter.cpp
index 1285c4ee..d884217e 100644
--- a/src/dxvk/dxvk_adapter.cpp
+++ b/src/dxvk/dxvk_adapter.cpp
@@ -427,7 +427,8 @@ namespace dxvk {
       m_deviceFeatures.khrPresentWait.presentWait;
 
     // Unless we're on an Nvidia driver where these extensions are known to be broken
-    if (matchesDriver(VK_DRIVER_ID_NVIDIA_PROPRIETARY, Version(), Version(535, 0, 0))) {
+    // under gamescope
+    if (matchesDriver(VK_DRIVER_ID_NVIDIA_PROPRIETARY)) {
       enabledFeatures.khrPresentId.presentId = VK_FALSE;
       enabledFeatures.khrPresentWait.presentWait = VK_FALSE;
     }
-- 
2.47.1

